src/framebuffer.rs

```diff
@@ -141,7 +140,7 @@ macro_rules! impl_bit {
                         let byte_index = bytes_per_row * y + (x / pixels_per_bit);
                         let bit_index = 8 - (x % pixels_per_bit + 1) * C::Raw::BITS_PER_PIXEL;
 
-                        let mask = !($raw_type::new(0xFF).into_inner() << bit_index);
+                        let mask = !((2u8.pow(C::Raw::BITS_PER_PIXEL as u32) - 1) << bit_index);
```

Comment:
Nice performance improvement for such a small change!

Reply:
Yeah! I was surprised that it would help at all, moreso how much.

Reply:
My guess would be that the compiler doesn't recognize that the value is constant because `into_inner` isn't a const fn.

But I'm currently working on the `tinytga` PR and did some benchmarking/optimizing myself and I was surprised many times how much/little some changes impacted the performance. Maybe compilers are magic and we shouldn't even try to understand them.

Reply:
> My guess would be that the compiler doesn't recognize that the value is constant because into_inner isn't a const fn.

That could be it, but it's hard to say. I tried to get some output from godbolt.org with the two approaches yesterday but didn't get anywhere; it either produced nothing (presumably optimised out) or way too much debug cruft so I couldn't see what the line was compiled to.